Update ToC in TaskRuns documentation

nbaudis opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- 
Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! 

In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind <kind>

Supported kinds are: bug, cleanup, design, documentation, feature, flake, misc, question, tep
-->

Updating the ToC in the `TaskRuns` documentation which had some headings missing. Additionally, the following smaller improvements were made:
- Fixed quote instead of backtick in ServiceAccount credentials heading
- Moved overview one level deeper to be consistent with the rest of the documentation (see [docs/pipelines.md](https://github.com/tektoncd/pipeline/blob/6d5217a956de56353ece80a5f6732caa86bfb0ee/docs/pipelines.md) and [docs/tasks.md](https://github.com/tektoncd/pipeline/blob/6d5217a956de56353ece80a5f6732caa86bfb0ee/docs/tasks.md))
- Added `<!-- toc -->` and `<!-- /toc -->` tags for inplace ToC updates in the future

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

<!--
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior
- Changes requiring upgrade notices or deprecation warnings

For pull requests with a release note:

```release-note
Your release note here
```

For pull requests that require additional action from users switching to the new release, include the string "action required" (case insensitive) in the release note:

```release-note
action required: your release note here
```

For pull requests that don't need to be mentioned at release time, use the `/release-note-none` Prow command to add the `release-note-none` label to the PR. You can also write the string "NONE" as a release note in your PR description:

```release-note
NONE
```
-->

```release-note
NONE
```

pritidesai commented:
/ok-to-test

pritidesai commented on docs/taskruns.md:
```diff
@@ -7,33 +7,39 @@ weight: 300
   - [Specifying `Sidecars`](#specifying-sidecars)
   - [Specifying `LimitRange` values](#specifying-limitrange-values)
-  - [Configuring the failure timeout](#configuring-the-failure-timeout)
+- [Configuring the failure timeout](#configuring-the-failure-timeout)
```
thanks @nbaudis for the updates 🙏 

This can be placed under Configuring a `TaskRun`. Please update the section header so that it aligns with the other "Specifying*" titles.

pritidesai commented on docs/taskruns.md:
```diff
@@ -7,33 +7,39 @@ weight: 300
-- [Debugging a `TaskRun`](#debugging-a-taskrun)
-- [Events](events.md#taskruns)
-- [Running a TaskRun Hermetically](hermetic.md)
+  - [Debugging a `TaskRun`](#debugging-a-taskrun)
```
Also, please move this out of the `Cancelling a TaskRun` section, such that, both are on same level:

Cancelling a TaskRun
Debugging a TaskRun

pritidesai commented:
There are two sections missing from ToC:

- [Events](events.md#taskruns)
- [Running a TaskRun Hermetically](hermetic.md)

Both are referencing other files, can we please add it back for folks if they are relying on those links.

nbaudis commented:
Thanks a lot for your comments @pritidesai!

I've corrected the indentation for the two sections and added the two links again. Maybe these links should be placed somewhere outside the ToC so they don't disappear again, but I'm not sure since maybe people rely on them being at the top of the documentation.

afrittoli commented:
/test pull-tekton-pipeline-alpha-integration-tests

sbwsg commented on docs/taskruns.md:
```diff
@@ -509,7 +517,7 @@ spec:
-### Debugging a `TaskRun`
+## Debugging a `TaskRun`
 
 #### Breakpoint on Failure
```
We should probably drop the heading on this to 3 `#` to bring it in line with its changed parent heading. Same for `#### Debug Environment` below - should now be `### Debug Environment`.

nbaudis replied:
Thanks @sbwsg, you are right. I've changed the levels of the two headings.

nbaudis commented:
squashed all commits into https://github.com/tektoncd/pipeline/commit/178976336b672edd99aa7fa822a69b370594de94

pritidesai reviewed (approved):
thank you @nbaudis 👍

sbwsg commented on docs/taskruns.md:
```diff
@@ -7,33 +7,41 @@ weight: 300
   - [Monitoring `Results`](#monitoring-results)
 - [Cancelling a `TaskRun`](#cancelling-a-taskrun)
 - [Debugging a `TaskRun`](#debugging-a-taskrun)
+    - [Breakpoint on Failure](#breakpoint-on-failure)
```
Looks like the toc needs to be re-generated after changing the heading level of these two. They're still indented one level too far I think.

pritidesai replied:
Hey @sbwsg I think it shows all aligned:

<img width="486" alt="Screen Shot 2021-09-29 at 6 30 18 AM" src="https://user-images.githubusercontent.com/206285/135278364-8fe37ded-b672-4f26-959a-d0822ce61c09.png">

pritidesai replied:
I see, the `- [Breakpoint on Failure](#breakpoint-on-failure)` is one level indented further compared to `- [Monitoring `Results`](#monitoring-results)`.

sbwsg replied:
Ah OK, looks like it renders fine even though the markdown spacing is different.

sbwsg commented:
/lgtm
